core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -135,6 +145,12 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
         TranslationKeyType.DESCRIPTION
       );
 
+    this.readOnlyExplorationBackendApiService
```

Comment:
The exploration data already exists in the page, presumably -- do you actually need to do a second independent request here?

Reply:
Hi @seanlip! I am making that request because I need the complete list of states of the exploration as well as the initial state, to then call the checkpointCelebrationUtilityService.getStateListForCheckpointMessages on line 230. I don't think this information is available on the previous page, but I can look into it.

I am following the example of what was done here in the use of that function:

https://github.com/oppia/oppia/blob/6345bf8893f67ee7bc80e615abf60cadb4707c61/core/templates/components/checkpoint-celebration-modal/checkpoint-celebration-modal.component.ts#L115

Reply:
Ah thanks. Yeah, both of these calls seem a bit weird to me. When the exploration player page loads it will get all the exploration data, so presumably this is stored somewhere to be retrievable by both the checkpoint modal and this one? Otherwise we keep making repeated fetches to the backend for the same data, which seems inefficient.

Reply:
Ah yes, that makes sense...I will try to understand where that information is so it can be passed to this page, and perhaps the other.